integrations/terraform-cloud/main.py

```diff
@@ -19,7 +24,10 @@ class ObjectKind(StrEnum):
 
 
 SKIP_WEBHOOK_CREATION = False
-SEMAPHORE_LIMIT = 30
+
+# Constants for rate limiting
+MAX_CONCURRENCY = 25  # Keep slightly under the 30/sec limit to allow for overhead
+semaphore = asyncio.BoundedSemaphore(MAX_CONCURRENCY)
```

Comment:
why you are defining it in two places? shouldn't client be enough?

Reply:
I have removed this from all the methods. 

Now, I still maintain the concurreny control in resync_runs by using `process_queue` because the API was timing out without any concurrent limit of how many workspaces are processed at a ago